documentation/docs/protocol_specification/components/congestion_control.md

```diff
@@ -1,87 +1,173 @@
-*   _Message dequeuing_. The first message of the queue is dequeued, and list of active nodes is updated.
-*   _Scheduler management_. Scheduler counters and pointers are updated.
+* __Classification__: The mechanism identifies the queue where the message belongs according to the node ID of
+  the message issuer.
+* __Message enqueuing__: The message is actually enqueued, the queue is sorted by message timestamps in increasing order
+  and counters are updated (e.g., counters for the total number of bytes in the queue).
+* __Message skipping__. A message is skipped when it is confirmed while still waiting to be scheduled in the
+  outbox. A skipped message is removed from the outbox buffer, and it does not decrease the issuing node's deficit.
+  It is not gossipped or added to the tip pool as confirmation indicates that the message already has approvers and
+  is replicated on enough nodes in the network.
+* __Message drop__: In some circumstances, due to network congestion or ongoing attacks, some messages shall be dropped
+  to guarantee bounded delays and isolate the attacker's messages. Specifically, a node shall drop messages in the
+  following situations:
+    - Since the total outbox buffer has a limited size, if the total number of bytes in all queues exceeds the current
+      buffer size is smaller than the maximum size. The Mana scaled length of a queue is calculated by dividing the number
+      of bytes in the queue by the amount of access Mana of the corresponding node:
+    - `Mana-scaled queue size = queue size / node aMana`;
+
+The dequeuing mechanism includes the following components:
+
+* __Queue selection__: A queue is selected according to a round-robin scheduling algorithm. In particular, the 
+mechanism uses a modified version of the deficit round-robin (DRR) algorithm.
+* __Message dequeuing__. The first (oldest) message of the queue, that satisfies certain conditions is dequeued. A 
+  message must satisfy the following conditions:
+    * The message has a ready flag assigned. Ready flag is assigned to a message when all of its parents are eligible (parents have been scheduled or confirmed).
+    * The message timestamp is not in the future.
+* __Scheduler management__: The scheduler counters and pointers are updated.
+
+#### False positive drop
+
+During an attack or congestion, a node may drop a message already scheduled by the rest of the network, causing a  
+_false positive drop_. This means that the message’s future cone will not be marked as _ready_ as its past cone is not
+eligible. This will be resolved when the rest of the network confirms the message, making it eligible and satisfying
+the _consistency_ requirement.
+
+#### False negative drop
+
+Another possible problem is that a node schedules a message that the rest of the network drops, causing a _false
```

Comment:
False negative drop is a strange term to me as we're actually not dropping something :)

Reply:
False positive schedule sounds better?